apps/dot-cdn/src/app/app.component.html

```diff
@@ -2,58 +2,91 @@
+                        ></p-skeleton>
+                        <ng-template #valuesBlock>
+                            <h3 class="dot-cdn__stats-value">{{ stats.value }}</h3>
+                        </ng-template>
+                        <dot-icon [name]="stats.icon" size="32"></dot-icon>
                     </div>
                 </div>
 
                 <div class="dot-cdn__chart">
-                    <h3 class="dot-cdn__tab-content-label">Bandwidth Served</h3>
-                    <p-chart #chart type="line" [data]="data" width="100%" height="50vh"></p-chart>
+                    <dot-spinner
+                        [size]="'4rem'"
+                        [borderSize]="'3px'"
+                        *ngIf="isLoading; else chartBlock"
+                        [ngStyle]="{ height: chartHeight }"
+                    ></dot-spinner>
+                    <ng-template #chartBlock>
+                        <h3 class="dot-cdn__tab-content-label">Bandwidth Served</h3>
+                        <p-chart
+                            #chart
+                            type="line"
+                            [data]="chartData"
+                            width="100%"
+                            [height]="chartHeight"
+                            [options]="options"
+                        ></p-chart>
+                    </ng-template>
                 </div>
             </div>
         </p-tabPanel>
         <p-tabPanel header="Flush Cache">
             <div class="dot-cdn__tab-content--contained">
-                <div class="dot-cdn__tab-content__row">
-                    <h3 class="dot-cdn__tab-content-label">Purge URL List</h3>
-                    <p>
-                        Purging an url list will remove the file from the CDN cache and re-download
-                        it from your server. Please enter the exact CDN URL of each individual file.
-                        Wildcard values are not supported.
-                    </p>
-                    <textarea [rows]="5" pInputTextarea></textarea>
-                    <button pButton label="Purge URL" disabled="true"></button>
-                </div>
-                <div class="dot-cdn__tab-content__row">
-                    <h3 class="dot-cdn__tab-content-label">Purge Pull Zone</h3>
-                    <p>
-                        Purging a pull zone will remove completely remove all of its files from
-                        cache and foce everything to be re-downloaded from your origin. If your zone
-                        uses a lot of storage, this ight send a large amount of traffic back to your
-                        server, so use with caution.
-                    </p>
-                    <button
-                        pButton
-                        label="Purge Pull Zone"
-                        class="p-button-danger p-button-outlined"
-                    ></button>
-                </div>
+                <form [formGroup]="purgeZoneForm">
```

Comment:
You don't need to wrap all this in the form, you need to wrap only the textarea and it button.